### PR TITLE
[SPARK-37979][SQL] Switch to more generic error classes in AES functions

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to switch from specific error classes in the AES functions: `aes_encrypt()/aes_decrypt()` to more generic:
- AES_CRYPTO_ERROR -> INVALID_PARAMETER_VALUE
- INVALID_AES_KEY_LENGTH -> INVALID_PARAMETER_VALUE
- UNSUPPORTED_AES_MODE -> UNSUPPORTED_FEATURE

The new error classes `INVALID_PARAMETER_VALUE` and `UNSUPPORTED_MODE` are made to be re-used from other functions but not only in the AES functions.

### Why are the changes needed?
1.  To prevent unlimited inflation of the set of error classes and as a consequence of that inflation of `error-classes.json`.
2. To establish rules for other sub-tasks of SPARK-37935

### Does this PR introduce _any_ user-facing change?
Yes, but the AES functions `aes_encrypt()`/`aes_decrypt()` haven't released yet.

### How was this patch tested?
By running the affected test suites:
```
$ build/sbt "test:testOnly *SparkThrowableSuite"
$ build/sbt "test:testOnly *QueryExecutionErrorsSuite"
```

### CODE REVIEW

**Comment by MaxGekk** [comment]
@cloud-fan @sarutak @srielau @entong Could you review this PR, please.

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -160,6 +153,10 @@
     "message" : [ "Unsupported literal type %s %s" ],
     "sqlState" : "0A000"
   },
+  "UNSUPPORTED_MODE" : {
```
Comment (cloud-fan) [inline core/src/main/resources/error/error-classes.json:156]: `mode` is also a parameter of the aes function, shall we reuse `INVALID_PARAMETER_VALUE`

File: core/src/main/resources/error/error-classes.json [hunk 2]
```diff
@@ -99,6 +92,10 @@
   "INVALID_JSON_SCHEMA_MAPTYPE" : {
     "message" : [ "Input schema %s can only contain StringType as a key type for a MapType." ]
   },
+  "INVALID_PARAMETER_VALUE" : {
+    "message" : [ "Invalid parameter value of '%s' in '%s'. %s" ],
```
Comment (cloud-fan) [inline core/src/main/resources/error/error-classes.json:96]: how about `The value of parameter '%s' in %s is invalid: %s`. Then we can have something like
`The value of parameter 'key' in function 'aes_encrypt/aes_decrypt' is invalid: expects a binary value with 16, 24 or 32 bytes, but got 9 bytes.`

File: core/src/main/resources/error/error-classes.json (hunk 1 above)
Comment (MaxGekk) [reply core/src/main/resources/error/error-classes.json:156]: Actually, any failure/error can be considered as a result of an invalid parameter. We could consider the entire universe as the input to Spark's code.

Comment (MaxGekk) [reply core/src/main/resources/error/error-classes.json:156]: @cloud-fan What is the use case for the sqlState = 0A000 (feature not supported, see https://github.com/apache/spark/tree/master/core/src/main/resources/error) from your point of view?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala [hunk 3]
```diff
@@ -1905,15 +1905,27 @@ object QueryExecutionErrors {
   }
 
   def invalidAesKeyLengthError(actualLength: Int): RuntimeException = {
-    new SparkRuntimeException("INVALID_AES_KEY_LENGTH", Array(actualLength.toString))
+    new SparkRuntimeException(
+      errorClass = "INVALID_PARAMETER_VALUE",
+      messageParameters = Array(
+        "key",
+        "aes_encrypt/aes_decrypt",
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:1912]:
```suggestion
        "function aes_encrypt/aes_decrypt",
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala [hunk 4]
```diff
@@ -1905,15 +1905,27 @@ object QueryExecutionErrors {
...
+        s"expects a binary value with 16, 24 or 32 bytes, but got ${actualLength.toString} bytes."))
   }
 
   def aesModeUnsupportedError(mode: String, padding: String): RuntimeException = {
-    new SparkRuntimeException("UNSUPPORTED_AES_MODE", Array(mode, padding))
+    new SparkRuntimeException(
+      errorClass = "UNSUPPORTED_FEATURE",
+      messageParameters = Array(s"AES-$mode with the padding $padding"))
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:1919]:
```suggestion
      messageParameters = Array(s"AES-$mode with the padding $padding in function aes_encrypt/aes_decrypt"))
```

**Review by cloud-fan — APPROVED** [review]

**Comment by MaxGekk** [comment]
Merging to master. Thank you, @cloud-fan .

**Pull request closed without merging**